### DISCUSSION TITLE
Conditional property binding

### DISCUSSION THREAD

**Discussion by HitecExports** [slint-ui/slint#3818] [Q&A] [answered]
Hello!

Slint supports nice ternary operator ... ? ... : ... for properties.

**Is there any way to have conditional property assignment?**

For example, 

`<property> := if <condition> is true then assign <value>, otherwise DO NOTHING (leave property untouched)`

I can't use ternary operator in this case (`<property> : <condition> ? <value> : <property>`),
because I will get error: "The binding for the `property `<property> is part of a binding loop"

I also can't use callback for evaluating conditional statement, because I need it in resizing event (As far as I know, Slint does not firing resizing callbacks).

The Conditional Element syntax (`if condition : id := Element { ... }`) also not working for me, because I need to update the property without creating any element

**Comment by bjorn** [comment, marked as answer]
Could it be a solution to use the condition to trigger a state, which then does the desired binding? See [an example here](https://slintpad.com/?snippet=import+%7B+Button%2C+VerticalBox+%7D+from+%22std-widgets.slint%22%3B%0Aexport+component+Demo+%7B%0A++++text-item+%3A%3D+Text+%7B%0A++++++++font-size%3A+24px%3B%0A++++++++horizontal-alignment%3A+center%3B%0A++++%7D%0A%0A++++states+%5B%0A++++++++bind-text+when+true%3A+%7B%0A++++++++++++text-item.text%3A+%22Hello+World%22%3B%0A++++++++%7D%0A++++%5D%0A%7D%0A):

```dart
import { Button, VerticalBox } from "std-widgets.slint";
export component Demo {
    text-item := Text {
        font-size: 24px;
        horizontal-alignment: center;
    }

    states [
        bind-text when true: {
            text-item.text: "Hello World";
        }
    ]
}
```

**Reply by HitecExports** [reply]
Thank you very much, Bjorn!

This technique really helps!

The only **question** is it possible to use value of the property in condition and in assignment like:

```
 states [
        bind-text when text-item.text !="Hello World" : {
            text-item.text: "Hello World";
        }
    ]
```

This will lead to the error: The binding for the property 'state' is part of a binding loop.

Using aliases (cached values) of the property also causing the same error.

I have tried functions. But they must be pure (without any changing of properties).

As I understand that checking for binding loop is used to prevent infinite recursion.

**Reply by bjorn** [reply]
I wonder what you want to happen? Regardless of the error, your code says "if it's not 'Hello World', bind it to 'Hello World'", but that means entering that state will immediately cause the condition for that state to become false again.

**Reply by HitecExports** [reply]
Yes, I just was thinking in imperative style of programming. I have refactored my code to avoid binding loop error.

Thank you very much for your help!
